FeedStoreChallenge/CoreDataFeedStore.swift

```
@@ -0,0 +1,122 @@
+//
+//  Created by Daniel Moro on 23.3.21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//  
+
+import CoreData
+
+public class CoreDataFeedStore: FeedStore {
+	
+	enum Error: Swift.Error {
+		case modelNotFound
+	}
+	
+	private var container: NSPersistentContainer
+	private var context: NSManagedObjectContext
+	private static var modelName: String {
+		"CoreDataFeedModel"
+	}
+	
+	public init(storeURL: URL, modelBundle: Bundle = Bundle.main) throws {
+		guard let modelURL = modelBundle.url(forResource: CoreDataFeedStore.modelName, withExtension: "momd"),
+			  let model = NSManagedObjectModel(contentsOf: modelURL) else {
+			throw Error.modelNotFound
+		}
+		
+		container = NSPersistentContainer(name: CoreDataFeedStore.modelName, managedObjectModel: model)
+		container.persistentStoreDescriptions = [NSPersistentStoreDescription(url: storeURL)]
+		container.loadPersistentStores(completionHandler: {_,_ in})
```

`loadPersistentStores` may fail, but the error is being ignored inside the closure with `_`.

If there's an error, it should be thrown so clients can handle it. Otherwise, it'll lead to runtime issues.

---

This error can be thrown later when the object is already initialized. I have thoughts to block this call with semaphore just to be able to fail if the error is raised but later I decided not to do that. Another option was to save error as a state in some enum:
```
enum containerState {
    case notLoaded
    case loading
    case loaded
    case failed(Error)
}
```
and then tho check this every time in `perform`

Do you have some suggestions?

---

@danielmoro the error shouldn't be thrown later because this object cannot work without a persistent store. It's unusable at runtime without one. So it shouldn't be possible to instantiate it with the persistent store.

You don't need to block the call because it's synchronous by default (even though it's closure-based). We show how to do it in the lectures - you can also check how other students did it in other pull requests ✅